src/Console/CacheDisposableDomainsCommand.php

```diff
@@ -20,72 +21,28 @@ class CacheDisposableDomainsCommand extends Command
      */
     protected $description = 'Caches the latest disposable email domains list';
 
-    /**
-     * The JSON source URL.
-     *
-     * @var string
-     */
-    public static $sourceUrl = 'https://rawgit.com/andreis/disposable-email-domains/master/domains.json';
-
     /**
      * Execute the console command.
      *
      * @return void
      */
     public function handle()
     {
-        if (! $data = $this->fetchFromSource()) {
-            $this->error('Couldn\'t reach the list source. Aborting.');
+        try {
 
-            return;
-        }
+            Indisposable::flushCache();
```

Comment:
Consider the following scenario:
1) Cache gets flushed
2) Remote domains are fetched, but fail
3) Even though there was a valid domains list in cache, you just flushed them away and are now forced to use the list shipped in the package instead.

-> Store remote domains on disk and use that one as a first fallback. Use the shipped package domains only as a final fallback if remote domains were never fetched.

Reply:
Oh, good catch!

The issue has been addressed with the new update, along with a test case replicating the scenario. 😃